plugins/catalog/src/api/index.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { createApiRef } from '@backstage/core';
+import { DescriptorEnvelope } from './types';
+
+export const catalogApiRef = createApiRef<CatalogApi>({
```

Comment:
Don't use the implementation itself as the type here, it makes it trickier to override and provide mock implementations.

Would also prefer to split this file up, to not have the implementation index. Something like `types.ts` with a`CatalogApi` type and the ApiRef, and `CatalogClient` that implements the API.

Reply:
Got it!